src/@types/prisma.ts

```diff
@@ -0,0 +1,57 @@
+export interface IPrismaSession {
+  id: string;
+  sid: string;
+  data: string | null;
+  expires: Date;
+}
+
+interface ICreatePrismaSession extends IPrismaSession {
+  data: string;
+}
+
+interface IFindOneArgs {
+  where: {
+    sid: string;
+  };
+  select?: {
+    expires?: boolean;
+    sid?: boolean;
+  };
+}
+
+interface IFindManyArgs {
+  where?: {
+    sid?: string;
+  };
+  select?: {
+    expires?: boolean;
+    sid?: boolean;
+    data?: boolean;
+  };
+}
+
+interface ICreateArgs {
+  data: ICreatePrismaSession;
+}
+
+interface IUpdateArgs {
+  where: { sid: string };
+  data: Partial<ICreatePrismaSession>;
+}
+
+interface IDeleteArgs {
+  where: { sid: string };
+}
+
+export interface IPrisma {
+  $connect(): Promise<void>;
+  $disconnect(): Promise<void>;
+  session: {
+    findOne(args: IFindOneArgs): Promise<IPrismaSession | null>;
+    findMany(args?: IFindManyArgs): Promise<IPrismaSession[]>;
+    delete(args?: IDeleteArgs): Promise<IPrismaSession>;
+    deleteMany(args?: any): Promise<any>;
```

Comment:
This is simply an interface of the `PrismaClient`. which does not provide a `deleteAll()`
Arguments are required here to match their interface otherwise a type error will occur when attempt to use a valid `PrismaClient`

![Screen Shot 2020-09-07 at 21 32 40](https://user-images.githubusercontent.com/8206108/92433365-afa9b080-f151-11ea-886c-e1360d2ac2d6.png)

Reply:
Got it. Seems unclear to me why PrismaClient's interface for `delete` and `deleteMany` have arguments that are _optional_, but yeah, makes sense to match their interface, and I imagine they have their reasons.